app/src/main/java/org/mozilla/fenix/share/listadapters/AccountDevicesAdapter.kt

```diff
@@ -23,25 +27,33 @@ class AccountDevicesShareAdapter(
         return AccountDeviceViewHolder(view, interactor)
     }
 
-    override fun getItemCount(): Int = devices.size
-
     override fun onBindViewHolder(holder: AccountDeviceViewHolder, position: Int) {
-        holder.bind(devices[position])
+        holder.bind(getItem(position))
     }
 
-    fun updateData(deviceOptions: List<SyncShareOption>) {
-        this.devices.clear()
-        this.devices.addAll(deviceOptions)
-        notifyDataSetChanged()
+    private object DiffCallback : DiffUtil.ItemCallback<SyncShareOption>() {
+        override fun areItemsTheSame(oldItem: SyncShareOption, newItem: SyncShareOption) =
+            when (oldItem) {
+                is SyncShareOption.SendAll -> newItem is SyncShareOption.SendAll
+                is SyncShareOption.SingleDevice ->
+                    newItem is SyncShareOption.SingleDevice && oldItem.device.id == newItem.device.id
+                else -> oldItem === newItem
+            }
+
+        @Suppress("DiffUtilEquals")
+        override fun areContentsTheSame(oldItem: SyncShareOption, newItem: SyncShareOption) =
+            oldItem == newItem
     }
 }
 
+/**
+ * Different options to be displayed by [AccountDevicesShareAdapter].
+ */
 sealed class SyncShareOption {
     object Reconnect : SyncShareOption()
     object Offline : SyncShareOption()
     object SignIn : SyncShareOption()
     object AddNewDevice : SyncShareOption()
     data class SendAll(val devices: List<Device>) : SyncShareOption()
-    data class Mobile(val name: String, val device: Device) : SyncShareOption()
-    data class Desktop(val name: String, val device: Device) : SyncShareOption()
+    data class SingleDevice(val device: Device) : SyncShareOption()
```

Comment:
Consider if this still works well when we introduce other devices like [VR, TV, etc](https://searchfox.org/mozilla-mobile/source/android-components/components/concept/sync/src/main/java/mozilla/components/concept/sync/Devices.kt#113).

Reply:
The intent here was to make it easier to introduce other devices